zerver/views/messages.py

```diff
@@ -153,9 +153,7 @@ def _pg_re_escape(self, pattern):
         s = list(pattern)
         for i, c in enumerate(s):
             if c not in self._alphanum:
-                if c == '\000':
```

Comment:
Quick note: We don't want this code path to ever 500/throw an exception, so if we go this route, we should throw a JsonableError or something here.

But looking at those tests above, is it possible that there's a way to change how we do encoding on the next line to make the tests still work?

If not, we might just want to ban null characters from stream names, so that this case can never happen.